pdns/libssl.cc

```diff
@@ -707,18 +721,74 @@ int OpenSSLTLSTicketKey::encrypt(unsigned char keyName[TLS_TICKETS_KEY_NAME_SIZE
     return -1;
   }
 
+#if OPENSSL_VERSION_MAJOR >= 3
+  using ParamsBuilder = std::unique_ptr<OSSL_PARAM_BLD, decltype(&OSSL_PARAM_BLD_free)>;
+
+  auto params_build = ParamsBuilder(OSSL_PARAM_BLD_new(), OSSL_PARAM_BLD_free);
+  if (params_build == nullptr) {
+    return -1;
+  }
+
+  if (OSSL_PARAM_BLD_push_utf8_string(params_build.get(), OSSL_MAC_PARAM_DIGEST, "sha256", 7) == 0) {
```

Comment:
It might be nice to move "sha256" to a constant, but more importantly, what is `7`?

Reply:
"The length of the string bsize should not include the terminating NUL byte." So it dos not look like the length of "sha256"

Reply:
You can also use 0 to let the function compute the length.

Reply:
I used 7 as the length of "sha256" and missed the part about not counting the terminating NUL byte. I also didn't want the function the re-compute something we already know the length of.

Does the following look OK?

```
constexpr static const char* sha256KeyName{"sha256"};
constexpr static const size_t sha256KeyNameLength = 6;
```

Reply:
Understood, I thought I had missed something! :-) Yes, the constants look good, thank you!